Quantum-Control-Applications/Superconducting/Single Fixed Transmon/DRAG_calibration.py

```diff
@@ -0,0 +1,136 @@
+            assign(state, I > ge_threshold)
+            save(state, state_st)
+            wait(cooldown_time, "resonator")
+
+            align()
+
+            play('y180' * amp(a, 0, 0, 1), 'qubit')
+            play('x90' * amp(1, 0, 0, a), 'qubit')
+            align("qubit", "resonator")
+            measure(
+                "readout",
+                "resonator",
+                None,
+                dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I),
+                dual_demod.full("rotated_minus_sin", "out1", "rotated_cos", "out2", Q),
+            )
+            save(I, I_st)
+            save(Q, Q_st)
+            assign(state, I > ge_threshold)
+            save(state, state2_st)
+            wait(cooldown_time, "resonator")
+        save(n, n_st)
+
+    with stream_processing():
+        I_st.buffer(len(amps)).average().save("I")
+        Q_st.buffer(len(amps)).average().save("Q")
+        n_st.save("iteration")
+        state_st.boolean_to_int().buffer(len(amps)).average().save("state")
+        state2_st.boolean_to_int().buffer(len(amps)).average().save("state2")
+
+#####################################
+#  Open Communication with the QOP  #
+#####################################
+qmm = QuantumMachinesManager(qop_ip)
+
+simulate = True
+
+if simulate:
+    simulation_config = SimulationConfig(duration=1000)  # in clock cycles
+    job = qmm.simulate(config, drag, simulation_config)
+    job.get_simulated_samples().con1.plot()
+
+else:
+    qm = qmm.open_qm(config)
+
+    job = qm.execute(drag)
+    res_handles = job.result_handles
+    I_handle = res_handles.get("I")
+    Q_handle = res_handles.get("Q")
+    iteration_handle = res_handles.get("iteration")
+    I_handle.wait_for_values(1)
+    Q_handle.wait_for_values(1)
+    iteration_handle.wait_for_values(1)
+    state_handle = res_handles.get('state')
+    state_handle.wait_for_values(1)
+    state2_handle = res_handles.get('state2')
+    state2_handle.wait_for_values(1)
+    next_percent = 0.1  # First time print 10%
+
+    def on_close(event):
+        event.canvas.stop_event_loop()
+        job.halt()
+
+    f = plt.figure()
+    f.canvas.mpl_connect("close_event", on_close)
+    print("Progress =", end=" ")
+
+    while res_handles.is_processing():
+        iteration = iteration_handle.fetch_all()
+        if iteration / n_avg > next_percent:
+            percent = 10 * round(iteration / n_avg * 10)  # Round to nearest 10%
+            print(f"{percent}%", end=" ")
+            next_percent = percent / 100 + 0.1  # Print every 10%
+
+    plt.cla()
+    I = I_handle.fetch_all()
+    Q = Q_handle.fetch_all()
+    state = state_handle.fetch_all()
+    state2 = state2_handle.fetch_all()
+    iteration = iteration_handle.fetch_all()
+    print(f"{round(iteration / n_avg * 100)}%")
+    plt.plot(amps * drag_coef, state, label='x180y90')
+    plt.plot(amps * drag_coef, state2, label='y180x90')
+    plt.xlabel("drag coef")
+    plt.legend()
+
```

Comment:
How do you do the post processing to find Drag_coeff?
Now drag_coef is 0 so the plotting won't work...

Reply:
you will get two different lines that intersect with each other and from there you will have a rough estimate of the  value-- the value is at the intersection point. 

Another poissiblity is to do analysis with fitting but I'm waiting for the fitting tools

Reply:
Then the default value for drag_coeff should be 1 or better, it should be set to 1 in the script and remain 0 in the config, otherwise the x-vector=amps * drag_coef will just be an array of zeros and so the plot won't work...
Indeed it would be nice to have the analysis which finds the optimal drag_coeff done as well so let's wait for the fitting tools

Reply:
thanks, I moved it 1 and then the config will be need to be changes as well